tests/reference/cpp-test_builtin_pow-56b3f92.stdout

```diff
@@ -0,0 +1,199 @@
+    assert (__lpython_overloaded_0__abs(__lpython_overloaded_7__pow(0.000000, 53) - 0.000000) < eps);
+    assert (__lpython_overloaded_0__pow(4, 2) == 16);
+    assert (__lpython_overloaded_0__abs(__lpython_overloaded_7__pow(-4235.000000, 52) - 394800380598526378720936476336799774273305305904443955996320177992404102454228192853661558132283280490733920647962082901303487965679010854404517306573035287122910924343151116372519789002752.000000) < eps);
+    c1 = __lpython_overloaded_9__complex(4, 5);
+    c1 = __lpython_overloaded_9__pow(c1, 4);
+}
+
+float __lpython_overloaded_0__abs(double x)
+{
+    double _lpython_return_variable;
+    if (x >= 0.000000) {
+        _lpython_return_variable = x;
+        return _lpython_return_variable;
+    } else {
+        _lpython_return_variable = -x;
+        return _lpython_return_variable;
+    }
+    return _lpython_return_variable;
+}
+
+int __lpython_overloaded_0__pow(int x, int y)
+{
+    int _lpython_return_variable;
+    _lpython_return_variable = std::pow(x, y);
+    return _lpython_return_variable;
+}
+
+long long __lpython_overloaded_1__pow(long long x, long long y)
+{
+    long long _lpython_return_variable;
+    _lpython_return_variable = std::pow(x, y);
+    return _lpython_return_variable;
+}
+
+float __lpython_overloaded_2__pow(float x, float y)
+{
+    float _lpython_return_variable;
+    _lpython_return_variable = std::pow(x, y);
+    return _lpython_return_variable;
+}
+
+float __lpython_overloaded_3__pow(double x, double y)
+{
+    double _lpython_return_variable;
+    _lpython_return_variable = std::pow(x, y);
+    return _lpython_return_variable;
+}
+
+float __lpython_overloaded_4__pow(int x, float y)
+{
+    float _lpython_return_variable;
+    _lpython_return_variable = std::pow((float)(x), y);
+    return _lpython_return_variable;
+}
+
+float __lpython_overloaded_5__pow(float x, int y)
+{
+    float _lpython_return_variable;
+    _lpython_return_variable = std::pow(x, (float)(y));
+    return _lpython_return_variable;
+}
+
+float __lpython_overloaded_6__pow(int x, double y)
+{
+    double _lpython_return_variable;
+    _lpython_return_variable = std::pow((float)(x), y);
+    return _lpython_return_variable;
+}
+
+float __lpython_overloaded_7__pow(double x, int y)
+{
+    double _lpython_return_variable;
+    _lpython_return_variable = std::pow(x, (float)(y));
+    return _lpython_return_variable;
+}
+
+int __lpython_overloaded_8__pow(bool x, bool y)
+{
+    int _lpython_return_variable;
+    if (y && !x) {
+        _lpython_return_variable = 0;
+        return _lpython_return_variable;
+    }
+    _lpython_return_variable = 1;
+    return _lpython_return_variable;
+}
+
+std::complex<float> __lpython_overloaded_9__complex(int x, int y)
+{
+    std::complex<float> _lpython_return_variable;
+    _lpython_return_variable = std::complex<double>(x) + std::complex<double>(y)*{0.000000, 1.000000};
```

Comment:
Will it compile? We should improve the C++ codegen for `complex` types.

Reply:
Does not compile because of `{.., ..}`. Fixed that.